src/Microsoft.Framework.Primitives/StringValues.cs

```diff
@@ -218,16 +246,65 @@ public static StringValues Concat(StringValues values1, StringValues values2)
             }
 
             var combined = new string[count1 + count2];
-            var index = 0;
-            foreach (var value in values1)
+            values1.CopyTo(combined, 0);
+            values2.CopyTo(combined, count1);
+            return new StringValues(combined);
+        }
+
+        public struct Enumerator : IEnumerator<string>
```

Comment:
The whole point of the struct enumerator is to avoid an enumerator when `foreach` over `StringValues` directly. As such, it (along with the public `GetEnumerator` method that returns it) cannot be private, otherwise it would result in a box allocation.